### PR TITLE
Improve autoupdate round 2

### PR DESCRIPTION
**Pull request by lldelisle**
Looking at the logs there are at least 2 things to fix:
- deal with cases when multiple PR titles match
- clean repo if the current PR has the same changes as after autoupdate.

### CODE REVIEW

File: .github/workflows/autoupdate.yml [hunk 1]
```diff
@@ -116,19 +116,19 @@ jobs:
             # We cannot use the -A option of `gh pr list` because the PR author changes with the Personal Access Token used,
             # so we filter on the 3rd output field (headRefName, i.e. remote_repository:branch).
             if gh_pr_list_out=$(gh pr list --search "head:$REPO" | grep -P "^\d+\t[^\t]+\tplanemo-autoupdate:$REPO\t"); then
-              OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2)
+              OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2 | head -n 1)
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:119]: I don't GitHub lets you open more than one pull request from the same branch.

```suggestion
              OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2)
```

File: .github/workflows/autoupdate.yml [hunk 2]
```diff
@@ -116,19 +116,19 @@ jobs:
...
               PR_EXISTS=1
-              PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1)
+              PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1 | head -n 1)
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:121]:
```suggestion
              PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1)
```

File: .github/workflows/autoupdate.yml [hunk 3]
```diff
@@ -116,19 +116,19 @@ jobs:
...
               PR_STATUS="OPENED"
               echo "PR exists and is opened"
               DIFF_BRANCH="origin/$REPO"
             else
               # Check if a closed PR exists with the same title
               if gh_pr_list_out=$(gh pr list --search "is:closed is:unmerged '$TITLE' in:title" | grep -P "^\d+\t[^\t]+\tplanemo-autoupdate:"); then
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:127]:
```suggestion
              if gh_pr_list_out=$(gh pr list --search "is:closed is:unmerged '$TITLE' in:title" | grep -P "^\d+\t[^\t]+\tplanemo-autoupdate:" | head -n 1); then
```

File: .github/workflows/autoupdate.yml [hunk 4]
```diff
@@ -116,19 +116,19 @@ jobs:
...
                 echo "Found a closed PR with title: $TITLE"
-                OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2)
+                OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2 | head -n 1)
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:129]:
```suggestion
                OLD_TITLE=$(echo "$gh_pr_list_out" | cut -f 2)
```

File: .github/workflows/autoupdate.yml [hunk 5]
```diff
@@ -116,19 +116,19 @@ jobs:
...
                 PR_EXISTS=1
-                PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1)
+                PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1 | head -n 1)
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:131]:
```suggestion
                PR_NUMBER=$(echo "$gh_pr_list_out" | cut -f 1)
```

**Comment by lldelisle** [comment]
@nsoranzo I added this `head -n 1` because you can have multiple closed PR from the same branch, see https://github.com/bgruening/galaxytools/pull/1353 and https://github.com/bgruening/galaxytools/pull/1350.
This is what caused the CI to fail for bgruening galaxytools repo: https://github.com/planemo-autoupdate/autoupdate/actions/runs/10995709180/job/30537933271#step:10:1069

But probably the `head -n 1` is only useful for closed PR.

**Comment by nsoranzo** [comment]
Yes, I think you missed my suggestion to add `| head -n 1` at line 127.

**Comment by lldelisle** [comment]
Indeed, I just fixed this.

**Comment by lldelisle** [comment]
Sorry...

File: .github/workflows/autoupdate.yml [hunk 6]
```diff
@@ -197,6 +197,10 @@ jobs:
                 echo "Creating a PR..."
                 gh pr create --base "${{ matrix.upstream_repo_branch }}" --head "planemo-autoupdate:$REPO" --title "$TITLE" --repo "${{ matrix.upstream_repo_owner}}/${{ matrix.upstream_repo_name }}" --body-file "${{ github.workspace }}/body.txt"
               fi
+            else
+              echo "No changes compared to current PR"
```
Comment (nsoranzo) [inline .github/workflows/autoupdate.yml:201]:
```suggestion
              echo "No changes compared to $DIFF_BRANCH"
```

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @lldelisle !

**Comment by lldelisle** [comment]
This will probably need a round 3 but thanks @nsoranzo for the review. I feel like we are moving forward.

**Pull request merged**